lib/matplotlib/patches.py

```diff
@@ -138,9 +138,16 @@ def contains(self, mouseevent, radius=None):
         if callable(self._contains):
             return self._contains(self, mouseevent)
         radius = self._process_radius(radius)
-        inside = self.get_path().contains_point(
-            (mouseevent.x, mouseevent.y), self.get_transform(), radius)
-        return inside, {}
+        codes = self.get_path().codes
+        vertices = self.get_path().vertices
+        idxs, = np.nonzero(codes == 1)
```

Comment:
Thanks for looking into this! What does this line do? Might be worth adding a comment just before it explaining.

Reply:
`np.where` might be clearer.